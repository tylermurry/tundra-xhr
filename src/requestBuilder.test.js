import buildRequest from './requestBuilder';

describe('requestBuilder', () => {
  let request;
  let response;

  beforeEach(() => {
    request = {
      _url: 'url',
      _headers: 'request headers',
      _method: 'request method',
      _body: 'request content',
    };

    response = {
      _headers: { 1: ['1'], 2: ['2'] },
      _status: 'response status',
      _body: 'response body',
    };
  });

  it('should build a standard request', () => {
    expect(buildRequest(request, response)).toMatchSnapshot();
  });

  it('should build a request without a request', () => {
    expect(buildRequest(null, response)).toMatchSnapshot();
  });

  it('should build a request without a response', () => {
    expect(buildRequest(request, null)).toMatchSnapshot();
  });
});
