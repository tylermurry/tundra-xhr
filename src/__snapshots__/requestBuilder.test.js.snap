// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requestBuilder should build a request without a request 1`] = `
Object {
  "request": null,
  "response": Object {
    "content": "response body",
    "headers": Object {
      "1": Array [
        "1",
      ],
      "2": Array [
        "2",
      ],
    },
    "statusCode": "response status",
  },
}
`;

exports[`requestBuilder should build a request without a response 1`] = `
Object {
  "request": Object {
    "content": "request content",
    "headers": "request headers",
    "method": "request method",
    "url": "url",
  },
  "response": null,
}
`;

exports[`requestBuilder should build a standard request 1`] = `
Object {
  "request": Object {
    "content": "request content",
    "headers": "request headers",
    "method": "request method",
    "url": "url",
  },
  "response": Object {
    "content": "response body",
    "headers": Object {
      "1": Array [
        "1",
      ],
      "2": Array [
        "2",
      ],
    },
    "statusCode": "response status",
  },
}
`;
